pkg/apis/batch/validation/validation_test.go

```diff
@@ -2138,9 +2249,393 @@ func TestValidateJobUpdateStatus(t *testing.T) {
+			enableManagedByLabel: true,
+			old: batch.Job{
+				ObjectMeta: validObjectMeta,
+			},
+			update: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Status: batch.JobStatus{
+					CompletionTime: &now,
+				},
+			},
+			wantErrs: field.ErrorList{
+				{Type: field.ErrorTypeInvalid, Field: "status.completionTime"},
+			},
+		},
+		"invalid CompletionTime when thre is no Complete condition; don't fail unrelated status change": {
+			enableManagedByLabel: true,
+			old: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Status: batch.JobStatus{
+					CompletionTime: &now,
+				},
+			},
+			update: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Status: batch.JobStatus{
+					CompletionTime: &now,
+					Active:         1,
+				},
+			},
+		},
+		"invalid attempt setting CompletedIndexes when non-indexed completion mode is used": {
+			enableManagedByLabel: true,
+			old: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Completions:    ptr.To[int32](5),
+					CompletionMode: completionModePtr(batch.NonIndexedCompletion),
+				},
+			},
+			update: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Completions:    ptr.To[int32](5),
+					CompletionMode: completionModePtr(batch.NonIndexedCompletion),
+				},
+				Status: batch.JobStatus{
+					CompletedIndexes: "0",
+				},
+			},
+			wantErrs: field.ErrorList{
+				{Type: field.ErrorTypeInvalid, Field: "status.completedIndexes"},
+			},
+		},
+		"invalid because CompletedIndexes set when non-indexed completion mode is used; don't fail unrelated status change": {
+			enableManagedByLabel: true,
+			old: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Completions:    ptr.To[int32](5),
+					CompletionMode: completionModePtr(batch.NonIndexedCompletion),
+				},
+				Status: batch.JobStatus{
+					CompletedIndexes: "0",
+				},
+			},
+			update: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Completions:    ptr.To[int32](5),
+					CompletionMode: completionModePtr(batch.NonIndexedCompletion),
+				},
+				Status: batch.JobStatus{
+					CompletedIndexes: "0",
+					Active:           1,
+				},
+			},
+		},
+		"invalid attempt setting FailedIndexes when not backoffLimitPerIndex": {
+			enableManagedByLabel: true,
+			old: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Completions:    ptr.To[int32](5),
+					CompletionMode: completionModePtr(batch.IndexedCompletion),
+				},
+			},
+			update: batch.Job{
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Completions:    ptr.To[int32](5),
+					CompletionMode: completionModePtr(batch.IndexedCompletion),
+				},
+				Status: batch.JobStatus{
+					FailedIndexes: ptr.To("0"),
+				},
+			},
+			wantErrs: field.ErrorList{
+				{Type: field.ErrorTypeInvalid, Field: "status.failedIndexes"},
+			},
+		},
+		"invalid attempt to set completedIndexes set for non-indexed job": {
```

Comment:
isn't this a duplicate of `invalid attempt setting CompletedIndexes when non-indexed completion mode is used`?

Reply:
Right, I deleted it now. I must have prepared a couple of "slots" for modifying  and forgot this is duplicated.